retrievalmarket/network/deal_stream.go

```diff
@@ -0,0 +1,60 @@
+package network
+
+import (
+	cborutil "github.com/filecoin-project/go-cbor-util"
+	"github.com/libp2p/go-libp2p-core/mux"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/filecoin-project/go-fil-components/retrievalmarket"
+)
+
+type DealStream struct {
```

Comment:
NB: I wonder if this type should be private to the module. is there any downside there?